### ISSUE TITLE
Strange audio issues in After Burner: Black Falcon

### ISSUE THREAD

**Issue by Panderner** [hrydgard/ppsspp#14010]
It seems there is a strange buzzing sound bug in Afterburner: Black Falcon.

Here's a video to hear the sound:

https://user-images.githubusercontent.com/37503397/106351779-3cc7ee80-6319-11eb-8d72-25a5417c177c.mp4

**Comment by hrydgard** [comment]
I guess this is not a new issue?

**Comment by Panderner** [comment]
Pause the game will stop the buzzing sound effects but when firing button is pressed after that bullet firing sound is looped temporarily until Pauses or fails the mission

https://user-images.githubusercontent.com/37503397/106352350-510dea80-631d-11eb-8e0a-9a242e801b03.mp4

**Comment by hrydgard** [comment]
might be some sceSas channel allocation issue or just a bug in looping, maybe we don't handle unaligned loop points or something. Weird that we haven't had this in other games...

I meant to ask if it's only happening in recent builds or if it goes way back. I would guess the latter...

**Comment by Panderner** [comment]
Isame happens for older versions of PPSSPP.

**Comment by unknownbrackets** [comment]
A log with SAS and ME set to VERBOSE or DEBUG would help.

-[Unknown]

**Comment by Panderner** [comment]
Here's some logs when i start the mission
```
06:04:986 AT3_Stream   I[ME]: HLE/sceAtrac.cpp:1696 0=sceAtracReleaseAtracID(0)
06:09:579 AT3_Stream   I[ME]: HLE/sceAtrac.cpp:1935 0=sceAtracSetDataAndGetID(08d083c4, 00004000): atrac3+ stereo audio
```

Unfortunately I can't find SAS logs yet.

**Comment by metehan989** [comment]
This happpened to me too. I was playing multiplayer. The sound was distorted at the beginning but it fixed itsself at midway/end of the mission. It seems that the game has loading issues. I am on ios and ppsspp version 1.11.3

**Comment by samosaphile** [comment]
I can confirm this issue still exists on a fairly recent build (1.11.3-742) using vulkan backend. Android. I tested with @metehan989. 

Also, don't wanna thread hijack but, would be nice if @ANR2ME could look at the netcode of this game. It's abit wierd. I'm not sure where to post but testing on low ports (1500, 2500, 10k don't seem to work at all for this game) I didn't test above 10k as I lost patience but 60k worked. It is strange, since most psp games work just fine.



> A log with SAS and ME set to VERBOSE or DEBUG would help.
> 
> -[Unknown]

Could someone please explain how to do this in layman terms? I'm too noob, if not I'll try to figure it out myself.

**Comment by ANR2ME** [comment]
> 
> 
> I can confirm this issue still exists on a fairly recent build (1.11.3-742) using vulkan backend. Android. I tested with @metehan989.
> 
> Also, don't wanna thread hijack but, would be nice if @ANR2ME could look at the netcode of this game. It's abit wierd. I'm not sure where to post but testing on low ports (1500, 2500, 10k don't seem to work at all for this game) I didn't test above 10k as I lost patience but 60k worked. It is strange, since most psp games work just fine (on low ports that is).
> 
> > A log with SAS and ME set to VERBOSE or DEBUG would help.
> > -[Unknown]
> 
> Could someone please explain how to do this in layman terms? I'm too noob, if not I'll try to figure it out myself.

Port offset is just a simple addition to the original port: `original_port + port_offset = used_port`, so if the game already use a high original port, adding more to it could ended getting a low port, since max port number is 65535 anything beyond it will wrapped back to 0 (ie. 65536 =0, 65537 = 1, so on).

Also, different games are using different port, some can use as low as port 1 (Port 0 = automatically/randomly assigned) originally, some already at a high ports.
There is also some reserved or already taken ports for background service on some OS, and bumping into these ports won't worked well/at all.

**Comment by unknownbrackets** [comment]
> Could someone please explain how to do this in layman terms? I'm too noob, if not I'll try to figure it out myself.

Under Developer Tools, there's a place you can set the logging channels.  You can set it per topic, in this case SAS and ME, which will avoid cluttering the log with unrelated noise.  Ideally, it should be set this way from the beginning of gameplay, and you shouldn't load a save state (in game saves are fine.)

On PC, git builds contain a batch file "InfoLog.bat" which will create a file containing all log data.  That's the easiest way.

-[Unknown]
